<a href="https://colab.research.google.com/github/adilsonsimiaoDK/StatisticToDS/blob/main/sampling.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [37]:
import pandas as pd
import random
import numpy as np


In [38]:
dataset = pd.read_csv('census.csv')

In [3]:
dataset.shape

(32561, 15)

Simple Random Sampling: 
Simple randon sampling you have select value of sampling that you define to work in my case I decided n = 100 value random, the function(sample of library pandas) return 100 values random,  if you want to work always with the same values config random_state=1.

In [5]:
dataset.tail()

age      workclass  final-weight  ... hour-per-week  native-country  income
32556   27        Private        257302  ...            38   United-States   <=50K
32557   40        Private        154374  ...            40   United-States    >50K
32558   58        Private        151910  ...            40   United-States   <=50K
32559   22        Private        201490  ...            20   United-States   <=50K
32560   52   Self-emp-inc        287927  ...            40   United-States    >50K

[5 rows x 15 columns]

In [39]:
df_srs = dataset.sample(n=100, random_state=1)


In [ ]:
df_srs.head(10)

In [24]:
def simple_random_sampling(dataset, sampling):
  return dataset.sample(n=sampling) 

In [25]:
df_srs = simple_random_sampling(dataset, 100)
df_srs.shape

(100, 15)

Systematic Sampling: 

1.   You have to get total rows with function shape of your dataset and divedid by number of sampling that you wan to work eg: my dataset has rows 32561, columns 15, you must get rows dataset/sampling or 32561//100 = 325 it will return a number int.
2. Yo must choose a value random 0 up to dataset in 325  



In [34]:

def systematicsampling(dataset, sampling):
  interval = len(dataset)//sampling
  random.seed(1)  
  valueF = random.randint(0, interval)
  indices = np.arange(valueF, len(dataset), step=interval)
  systematic_sampling = dataset.iloc[indices]
  return systematic_sampling



In [ ]:
print(systematicsampling(df_srs,100))

Cluster for Sampling:
1. You have divide the dataset by number de sampling after that will be created groups that you should identify for id, 
2. command used: np.unique(group, return_counts=True) return the values not duplicate and count of element in each group.


In [40]:
def cluster_sampling(dataset, number_group):
  group = []
  group_id = 0
  count = 0
  interval = len(dataset)/number_group
  for _ in dataset.iterrows():
      group.append(group_id)
      count += 1 
      if count > interval:
        count = 0
        group_id +=1
  dataset['group'] = group
  selected_group = random.randint(0, number_group)
  return dataset[dataset['group']== selected_group]
  


In [50]:
group_sampling = cluster_sampling(dataset, 10)
group_sampling.shape, group_sampling.group.value_counts()
print(group_sampling)

       age          workclass  final-weight  ...  native-country  income group
13028   56            Private         98418  ...   United-States   <=50K     4
13029   30            Private        381153  ...   United-States   <=50K     4
13030   54            Private        103700  ...   United-States   <=50K     4
13031   35            Private        298635  ...   United-States   <=50K     4
13032   32            Private        127895  ...   United-States   <=50K     4
...    ...                ...           ...  ...             ...     ...   ...
16280   27            Private        187981  ...   United-States   <=50K     4
16281   55            Private        393768  ...   United-States   <=50K     4
16282   38            Private        108726  ...   United-States    >50K     4
16283   31            Private        180551  ...   United-States    >50K     4
16284   51   Self-emp-not-inc        176240  ...   United-States    >50K     4

[3257 rows x 16 columns]
